pkg/distributor/compat_stringlabels.go

```diff
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+//go:build stringlabels
+
+package distributor
+
+import (
+	"sync"
+
+	"github.com/prometheus/prometheus/model/labels"
+
+	"github.com/grafana/mimir/pkg/mimirpb"
+)
+
+// activeSeriesResponse is a helper to merge/deduplicate ActiveSeries responses from ingesters.
+type activeSeriesResponse struct {
+	m      sync.Mutex
+	series map[labels.Labels]struct{}
```

Comment:
I'm very surprised this works.
Not sure why you have two versions, but the standard way to get a map key from `Labels` is to call `.Bytes()`.

Reply:
My thinking was this: The spec says the requirement for map key types is to be comparable. Struct types are comparable if all their field types are comparable. `labels.Labels` has a single field of type `string`, so a `map[labels.Labels]struct{}` should be equivalent to a `map[string]struct{}`.

Do you think it would be preferable to use a byte slice as the map key here?